dbms/src/Debug/dbgFuncCoprocessor.cpp

```diff
@@ -645,7 +664,7 @@ QueryFragments mppQueryToQueryFragments(
     {
         mpp_ctx->sender_target_task_ids = current_task_ids;
         auto sub_fragments = mppQueryToQueryFragments(exchange.second.second, executor_index, properties, false, mpp_ctx);
-        receiver_source_task_ids_map[exchange.first] = sub_fragments.cbegin()->task_ids;
+        receiver_source_task_ids_map[exchange.first] = sub_fragments[sub_fragments.size() - 1].task_ids;
```

Comment:
The exchangeSender will have multiple sub_fragments if it is the parent of another exchangeSender. The last fragment of the sub_fragment is the exchangeSender itself.